experimental/packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -114,16 +126,20 @@ export class NodeSDK {
   }
 
   /** Set configurations needed to register a MeterProvider */
-  public configureMeterProvider(reader: MetricReader): void {
-    this._metricReader = reader;
+  public configureMeterProvider(config: MeterProviderConfig): void {
+    if (config?.views && !config.reader) {
+      throw new Error('A list of views have been passed but the NodeSDK expects that the MeterProvider is manually instantiated and can\'t attach Views to the MeterProvider');
```

Comment:
Sorry, I may not follow the point of the description of this error. Why does it require metric readers to present in the config to configure the views?

Reply:
You can’t pass views when you use your own meter provider as the views are part of the MeterProvider’s constructor. Hence throwing an error. Do you have a better suggestion for the wording?

Reply:
If I read it correctly, the if-statement checks the `config.reader`, not a "provider". They are different things.

Reply:
I think I maybe miscommunicated earlier. :sweat_smile: 

I meant that we need to handle the following issues:

- user passes `config.views` to the constructor, but also calls `configureMeterProvider()` and passes `config.views` 
- OR user passes `config.views` to `configureMeterProvider()` twice. 
 
Here, we have a few options:
  - use only the new ones and log a warning ("Overriding existing views with views passed to this method.")
  - throw an error ("Views already provided.")
  - combine both view lists ("Views already exist, adding provided views to existing.")

Reply:
I didn't even notice you could call `configureMeterProvider` separately

Reply:
I was thinking of something like this:

```ts
  /** Set configurations needed to register a MeterProvider */
  public configureMeterProvider(config: MeterProviderConfig): void {
	// nothing is set yet, we can set config and return.
    if (this._meterProviderConfig == null) {
      this._meterProviderConfig = config;
      return;
    }

    // make sure we do not override existing views with other views.
    if (this._meterProviderConfig.views != null && config.views != null) {
      throw new Error('Views passed but Views have already been configured.');
    }

    // set views, but make sure we do not override existing views with null/undefined.
    if (config.views != null){
      this._meterProviderConfig.views = config.views;
    }

    // make sure we do not override existing reader with another reader.
    if (this._meterProviderConfig.reader != null && config.reader != null) {
      throw new Error('MetricReader passed but MetricReader has already been configured.');
    }

    // set reader, but make sure we do not override existing reader with null/undefined.
    if (config.reader != null){
      this._meterProviderConfig.reader = config.reader;
    }
  }
```

It should be okay to have views but not a reader. Having that is a valid config. :slightly_smiling_face:

Reply:
Cool, I will take over this :)

Reply:
@weyert Sorry I noticed that you marked this as resolved but I still find this checks on the presence of view and reader exists. I didn't find @pichlermarc's suggestion has this check. Would you mind elaborating on this check why do we need this?

Reply:
@legendecas I am not sure, anymore, to tell you the truth. The checks by @pichlermarc make sense to me and I have added them and added tests for this.